exp/runtime/topologymutation/variables_test.go

```diff
@@ -82,51 +88,197 @@ func Test_GetStringTemplateVariable(t *testing.T) {
 
 	varA := apiextensionsv1.JSON{Raw: toJSON("a")}
 	tests := []struct {
-		name          string
-		variables     map[string]apiextensionsv1.JSON
-		variableName  string
-		expectedValue string
-		expectedFound bool
-		expectedErr   bool
+		name                  string
+		variables             map[string]apiextensionsv1.JSON
+		variableName          string
+		expectedValue         string
+		expectedNotFoundError bool
+		expectedErr           bool
 	}{
 		{
-			name:          "Fails for invalid variable reference",
-			variables:     nil,
-			variableName:  "invalid[",
-			expectedValue: "",
-			expectedFound: false,
-			expectedErr:   true,
+			name:                  "Fails for invalid variable reference",
+			variables:             nil,
+			variableName:          "invalid[",
+			expectedValue:         "",
+			expectedNotFoundError: false,
+			expectedErr:           true,
 		},
 		{
-			name:          "variable not found",
-			variables:     nil,
-			variableName:  "notEsists",
-			expectedValue: "",
-			expectedFound: false,
-			expectedErr:   false,
+			name:                  "variable not found",
+			variables:             nil,
+			variableName:          "notEsists",
+			expectedValue:         "",
+			expectedNotFoundError: true,
+			expectedErr:           true,
 		},
 		{
-			name: "variable not found",
+			name: "valid variable",
 			variables: map[string]apiextensionsv1.JSON{
 				"a": varA,
 			},
-			variableName:  "a",
-			expectedValue: "a",
-			expectedFound: true,
-			expectedErr:   false,
+			variableName:          "a",
+			expectedValue:         "a",
+			expectedNotFoundError: false,
+			expectedErr:           false,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			value, found, err := GetStringVariable(tt.variables, tt.variableName)
+			value, err := GetStringVariable(tt.variables, tt.variableName)
 
 			g.Expect(value).To(Equal(tt.expectedValue))
-			g.Expect(found).To(Equal(tt.expectedFound))
 			if tt.expectedErr {
 				g.Expect(err).To(HaveOccurred())
 			} else {
 				g.Expect(err).ToNot(HaveOccurred())
 			}
+
+			if tt.expectedNotFoundError {
+				g.Expect(patchvariables.IsNotFoundError(err)).To(BeTrue())
+			}
+		})
+	}
+}
+
+func Test_GetBoolVariable(t *testing.T) {
+	g := NewWithT(t)
+
+	varA := apiextensionsv1.JSON{Raw: toJSON("true")}
```

Comment:
```suggestion
	varA := apiextensionsv1.JSON{Raw: []byte(`true`)}
```

Bool should look like this in an apiextensionsv1.JSON (I looked at some of our unit tests in internal/topology/variables/cluster_variable_defaulting_test.go and I think I tested this when I wrote them)